src/vs/workbench/api/common/positron/extHostLanguageRuntime.ts

```diff
@@ -569,7 +569,31 @@ export class ExtHostLanguageRuntime implements extHostProtocol.ExtHostLanguageRu
 		if (handle >= this._runtimeSessions.length) {
 			throw new Error(`Cannot interrupt runtime: session handle '${handle}' not found or no longer valid.`);
 		}
-		return this._runtimeSessions[handle].interrupt();
+		const session = this._runtimeSessions[handle];
+		try {
+			session.interrupt();
+		} finally {
+			// Whether or not the interrupt was successful, ensure that
+			// execution observers associated with this session are settled, so
+			// that interrupting the session is always successful from the
+			// perspective of the observer even if the underlying session fails
+			// to interrupt.
+			this._executionObservers.forEach((observer, id) => {
+				if (observer.sessionId === session.metadata.sessionId) {
+					// The observer is associated with this session, so we
+					// need to clean it up. Reject its promise if it hasn't
+					// already been settled.
+					if (!observer.promise.isSettled) {
+						observer.onFailed({
```

Comment:
With this change, the observer will totally bail on the execution as soon as an interrupt is requested. That might be what we want for Assistant but I could see other users expecting the execution to complete through to the interrupt error and status idle.

It might also be possible that an execution completes successfully but the observer still sees it as failed due to interrupt. For example, if there is an in-flight status idle message associated with an execute request when the user interrupts.

What do you think about moving this to a `token.onCancellationRequested` listener in the `executeCode` tool? We could also leave it as is for now and deal with this if/when a usecase actually arises.

Reply:
That's a good question and I could see arguments for both sides. Interruption does not work on Python + Windows (https://github.com/posit-dev/positron/issues/4604) so the question of how to handle the case where interrupt never returns isn't academic.

I think that it's okay to always treat a user interruption as a failure from the perspective of the execution observer; even if the interrupt "succeeds" the computation did not.

Reply:
That sounds reasonable!